### ISSUE TITLE
storages module not found on heroku

### ISSUE THREAD

**Issue by morenoh149** [pydanny/cookiecutter-django#2328]
## What happened?
I follow the instructions for deploying to heroku. When I open the heroku app I see "Internal Server Error". When I review the docs I see
```
2019-11-25T18:32:18.934502+00:00 app[web.1]: return self.render(context)
2019-11-25T18:32:18.934504+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.7/site-packages/django/template/loader_tags.py", line 150, in render                                                                                    
2019-11-25T18:32:18.934506+00:00 app[web.1]: return compiled_parent._render(context)
2019-11-25T18:32:18.934508+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.7/site-packages/django/template/base.py", line 163, in _render                                                                                          
2019-11-25T18:32:18.934510+00:00 app[web.1]: return self.nodelist.render(context)
2019-11-25T18:32:18.934512+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.7/site-packages/django/template/base.py", line 937, in render                                                                                           
2019-11-25T18:32:18.934514+00:00 app[web.1]: bit = node.render_annotated(context)
2019-11-25T18:32:18.934516+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.7/site-packages/django/template/base.py", line 904, in render_annotated                                                                                 
2019-11-25T18:32:18.934518+00:00 app[web.1]: return self.render(context)
2019-11-25T18:32:18.934520+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.7/site-packages/django/template/loader_tags.py", line 62, in render                                                                                     
2019-11-25T18:32:18.934522+00:00 app[web.1]: result = block.nodelist.render(context)
2019-11-25T18:32:18.934524+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.7/site-packages/django/template/base.py", line 937, in render                                                                                           
2019-11-25T18:32:18.934526+00:00 app[web.1]: bit = node.render_annotated(context)
2019-11-25T18:32:18.934527+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.7/site-packages/django/template/base.py", line 904, in render_annotated                                                                                 
2019-11-25T18:32:18.934529+00:00 app[web.1]: return self.render(context)
2019-11-25T18:32:18.934531+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.7/site-packages/compressor/templatetags/compress.py", line 142, in render                                                                               
2019-11-25T18:32:18.934533+00:00 app[web.1]: return self.render_compressed(context, self.kind, self.mode, forced=forced)                                                                                                                   
2019-11-25T18:32:18.934535+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.7/site-packages/compressor/templatetags/compress.py", line 110, in render_compressed                                                                    
2019-11-25T18:32:18.934537+00:00 app[web.1]: cache_key, cache_content = self.render_cached(compressor, kind, mode)
2019-11-25T18:32:18.934539+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.7/site-packages/compressor/templatetags/compress.py", line 88, in render_cached                                                                         
2019-11-25T18:32:18.934541+00:00 app[web.1]: cache_key = get_templatetag_cachekey(compressor, mode, kind)
2019-11-25T18:32:18.934543+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.7/site-packages/compressor/cache.py", line 104, in get_templatetag_cachekey                                                                             
2019-11-25T18:32:18.934545+00:00 app[web.1]: "templatetag.%s.%s.%s" % (compressor.cachekey, mode, kind))
2019-11-25T18:32:18.934547+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.7/site-packages/django/utils/functional.py", line 80, in __get__                                                                                        
2019-11-25T18:32:18.934549+00:00 app[web.1]: res = instance.__dict__[self.name] = self.func(instance)
2019-11-25T18:32:18.934551+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.7/site-packages/compressor/base.py", line 200, in cachekey                                                                                              
2019-11-25T18:32:18.934553+00:00 app[web.1]: [self.content] + self.mtimes).encode(self.charset), 12)
2019-11-25T18:32:18.934555+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.7/site-packages/django/utils/functional.py", line 80, in __get__                                                                                        
2019-11-25T18:32:18.934557+00:00 app[web.1]: res = instance.__dict__[self.name] = self.func(instance)
2019-11-25T18:32:18.934559+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.7/site-packages/compressor/base.py", line 194, in mtimes                                                                                                
2019-11-25T18:32:18.934561+00:00 app[web.1]: for kind, value, basename, elem in self.split_contents()
2019-11-25T18:32:18.934563+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.7/site-packages/compressor/css.py", line 18, in split_contents                                                                                          
2019-11-25T18:32:18.934565+00:00 app[web.1]: basename = self.get_basename(elem_attribs['href'])
2019-11-25T18:32:18.934567+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.7/site-packages/compressor/base.py", line 95, in get_basename                                                                                           
2019-11-25T18:32:18.934569+00:00 app[web.1]: base_url = self.storage.base_url
2019-11-25T18:32:18.934571+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.7/site-packages/django/utils/functional.py", line 256, in inner                                                                                         
2019-11-25T18:32:18.934573+00:00 app[web.1]: self._setup()
2019-11-25T18:32:18.934574+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.7/site-packages/compressor/storage.py", line 120, in _setup                                                                                             
2019-11-25T18:32:18.934576+00:00 app[web.1]: self._wrapped = get_storage_class(settings.COMPRESS_STORAGE)()
2019-11-25T18:32:18.934578+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.7/site-packages/django/core/files/storage.py", line 358, in get_storage_class                                                                           
2019-11-25T18:32:18.934581+00:00 app[web.1]: return import_string(import_path or settings.DEFAULT_FILE_STORAGE)
2019-11-25T18:32:18.934583+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.7/site-packages/django/utils/module_loading.py", line 17, in import_string                                                                              
2019-11-25T18:32:18.934585+00:00 app[web.1]: module = import_module(module_path)
2019-11-25T18:32:18.934587+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.7/importlib/__init__.py", line 127, in import_module                                                                                                    
2019-11-25T18:32:18.934589+00:00 app[web.1]: return _bootstrap._gcd_import(name[level:], package, level)
2019-11-25T18:32:18.934596+00:00 app[web.1]: File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
2019-11-25T18:32:18.934598+00:00 app[web.1]: File "<frozen importlib._bootstrap>", line 983, in _find_and_load
2019-11-25T18:32:18.934600+00:00 app[web.1]: File "<frozen importlib._bootstrap>", line 953, in _find_and_load_unlocked                                                                                                                    
2019-11-25T18:32:18.934602+00:00 app[web.1]: File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed                                                                                                                  
2019-11-25T18:32:18.934604+00:00 app[web.1]: File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
2019-11-25T18:32:18.934606+00:00 app[web.1]: File "<frozen importlib._bootstrap>", line 983, in _find_and_load
2019-11-25T18:32:18.934608+00:00 app[web.1]: File "<frozen importlib._bootstrap>", line 953, in _find_and_load_unlocked                                                                                                                    
2019-11-25T18:32:18.934610+00:00 app[web.1]: File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed                                                                                                                  
2019-11-25T18:32:18.934614+00:00 app[web.1]: File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
2019-11-25T18:32:18.934616+00:00 app[web.1]: File "<frozen importlib._bootstrap>", line 983, in _find_and_load
2019-11-25T18:32:18.934618+00:00 app[web.1]: File "<frozen importlib._bootstrap>", line 965, in _find_and_load_unlocked                                                                                                                    
2019-11-25T18:32:18.934620+00:00 app[web.1]: ModuleNotFoundError: No module named 'storages'
2019-11-25T18:32:18.934622+00:00 app[web.1]: 10.109.218.102 - - [25/Nov/2019:18:32:18 +0000] "GET / HTTP/1.1" 500 0 "-" "-"     
```
pip freeze
```
$ heroku run pip freeze
Running pip freeze on ⬢ foobar... up, run.6133 (Hobby)
amqp==2.5.2
argon2-cffi==19.2.0
Babel==2.7.0
billiard==3.6.1.0
celery==4.3.0
certifi==2019.9.11
cffi==1.13.2
chardet==3.0.4
coreapi==2.3.3
coreschema==0.0.4
defusedxml==0.6.0
Django==2.2.7
django-allauth==0.40.0
django-anymail==7.0.0
django-appconf==1.0.3
django-celery-beat==1.5.0
django-compressor==2.3
django-crispy-forms==1.8.0
django-environ==0.4.5
django-model-utils==3.2.0
django-redis==4.10.0
django-timezone-field==3.1
djangorestframework==3.10.3
flower==0.9.3
gunicorn==20.0.0
idna==2.8
importlib-metadata==0.23
itypes==1.1.0
Jinja2==2.10.3
kombu==4.6.6
MarkupSafe==1.1.1
more-itertools==7.2.0
oauthlib==3.1.0
Pillow==6.2.1
psycopg2==2.8.3
pycparser==2.19
python-crontab==2.4.0
python-dateutil==2.8.1
python-slugify==4.0.0
python3-openid==3.1.0
pytz==2019.3
rcssmin==1.0.6
redis==3.3.11
requests==2.22.0
requests-oauthlib==1.3.0
rjsmin==1.1.0
six==1.13.0
sqlparse==0.3.0
text-unidecode==1.3
tornado==5.1.1
uritemplate==3.0.0
urllib3==1.25.7
vine==1.3.0
whitenoise==4.1.4
zipp==0.6.0
You are using pip version 9.0.2, however version 19.3.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
```

When I log in with `heroku run bash` and run `pip list` Django-compressor is listed correctly.

## What should've happened instead?
I should have seen the home page same as in local development

## Steps to reproduce

[//]: run app locally using custom bootstrap compilation and celery
[//]: deploy app using heroku cli and these instructions https://cookiecutter-django.readthedocs.io/en/latest/deployment-on-heroku.html
[//]: skip the aws steps (heroku config:set DJANGO_AWS_ACCESS_KEY_ID=) and the other aws configs
[//]: be sure to run the celery steps (celery_broker_url and ps:scale worker=1)

**Comment by morenoh149** [comment]
I got the site loading by commenting out https://github.com/pydanny/cookiecutter-django/blob/master/%7B%7Bcookiecutter.project_slug%7D%7D/config/settings/production.py#L203 and relying on the default (local filesystem) https://django-compressor.readthedocs.io/en/latest/settings/#django.conf.settings.COMPRESS_STORAGE

This is ok for my current needs. I want to run the app on heroku and serve the static files from there, not s3 for now.

**Comment by luzfcb** [comment]
When generating a new project, select 
`cloud_provider: None`, 
and 
`use_whitenoise: True`

**Comment by morenoh149** [comment]
I think I did make those choices, no cloud, and use white noise.

**Comment by luzfcb** [comment]
@morenoh149 I reread your report, and now I see where the error is 
https://github.com/pydanny/cookiecutter-django/blob/master/%7B%7Bcookiecutter.project_slug%7D%7D/config/settings/production.py#L203


I haven't tested this, but maybe it can be solved by replacing the lines [202 and 203](https://github.com/pydanny/cookiecutter-django/blob/9dc01f863ec7e421f09970ef04874027c96b1a73/%7B%7Bcookiecutter.project_slug%7D%7D/config/settings/production.py#L202-L203) with something like:


```jinja
# https://django-compressor.readthedocs.io/en/latest/settings/#django.conf.settings.COMPRESS_STORAGE
{% if cookiecutter.cloud_provider == 'AWS' -%}
COMPRESS_STORAGE = "config.settings.production.StaticRootS3Boto3Storage"
{%- elif cookiecutter.cloud_provider == 'GCP' -%}
COMPRESS_STORAGE = "config.settings.production.StaticRootGoogleCloudStorage"
{%- else -%}
COMPRESS_STORAGE = "compressor.storage.GzipCompressorFileStorage"
{%- endif %}
```

**Comment by morenoh149** [comment]
should I submit a pr with this fix?

**Comment by luzfcb** [comment]
@morenoh149 Yes, If you want and have free time to test if the configuration works as expected.
Anyway, thanks for reporting this issue.

**Comment by browniebroke** [comment]
I confirm that Fábio is right, the error was the combination of `use_compressor="y"` with `cloud_provider=None`. This has since been fixed in #2442 so I'm closing this.
